texlive's context program hangs on scanning of `/nix/store` dir

## Issue description

Seem to never finish its search. It seems wrong that this tool should search
such global location. It is as tough a recursive search from child to parent
dir is ongoing. Maybe an environment variable is not set properly? Maybe
I should add other packages in my `texlive.combine` function? 

As you can see, everything in `context`'s wrapper seem in order:

~~~
#! /nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/bash -e
export PATH=/nix/store/4kxg8pscnpcikkb8g77injdpwnbivp92-texlive-combined-2015/bin:/nix/store/s5hmy8kv1w151a1z4wy57csla3c11jr9-perl-5.20.3/bin${PATH:+:}$PATH
export TEXMFCNF=/nix/store/4kxg8pscnpcikkb8g77injdpwnbivp92-texlive-combined-2015/share/texmf/web2c
export TEXMFDIST=/nix/store/4kxg8pscnpcikkb8g77injdpwnbivp92-texlive-combined-2015/share/texmf
export TEXMFSYSCONFIG=/nix/store/4kxg8pscnpcikkb8g77injdpwnbivp92-texlive-combined-2015/share/texmf-config
export TEXMFSYSVAR=/nix/store/4kxg8pscnpcikkb8g77injdpwnbivp92-texlive-combined-2015/share/texmf-var
export PERL5LIB=/nix/store/4kxg8pscnpcikkb8g77injdpwnbivp92-texlive-combined-2015/share/texmf/scripts/texlive${PERL5LIB:+:}$PERL5LIB
exec /nix/store/pnc8xc1dh3cy4migfipfmzy3ad7snz99-bash-4.3-p42/bin/sh  /nix/store/ijr41d88sj8bqhd6bbqldwrh83b482qk-texlive-context-2015/scripts/context/stubs/unix/context "${extraFlagsArray[@]}" "$@"
~~~

Could reproduce on both stable (02/02/2016, e83aed4c10b05339329f4ca13414c59036563d0e)
and unstable (27/03/2016, d88ae1045d37f8cd1b3428617670f7358eb45a29).

### Steps to reproduce

The data:

~~~
git clone https://github.com/mszep/pandoc_resume
cd pandoc_resume
~~~

The env:

~~~
nix-shell --pure -E 'with import <nixpkgs> { }; stdenv.mkDerivation {name = "otf"; buildInputs = [pandoc (texlive.combine { inherit (texlive) scheme-small context cm-super; })];}'
~~~

First thing out of order: 

~~~
$ context --version

mtx-context     | ConTeXt Process Management 0.60
mtx-context     |
mtx-context     | main context file: unknown, 'context.mkiv' not found
~~~

And now the hanging scan issue:

~~~
pandoc --standalone --template style_chmduquesne.tex \
--from markdown --to context \
-V papersize=A4 \
-o resume.tex resume.md; \
context resume.tex

mtx-context     | warning: no format found, forcing remake (commandline driven)
resolvers       | resolving | configuration files already identified
resolvers       | resolving | loading configuration file '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf/web2c/texmfcnf.lua'
resolvers       | resolving |
resolvers       | resolving | locating list of 'home:.texlive2014/texmf-config' (runtime) (tree:///home:.texlive2014/texmf-config)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'tree', argument 'tree:///home:.texlive2014/texmf-config'
resolvers       | trees | locator 'home:.texlive2014/texmf-config' not found
resolvers       | resolving | locating list of 'home:texmf' (runtime) (tree:///home:texmf)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'tree', argument 'tree:///home:texmf'
resolvers       | trees | locator 'home:texmf' not found
resolvers       | resolving | locating list of '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-config' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-config'
resolvers       | files | file locator '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-config' not found
resolvers       | resolving | locating list of '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-var' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-var'
resolvers       | files | file locator '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-var' found as '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-var'
resolvers       | resolving | locating list of 'selfautoparent:texmf-project' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument 'selfautoparent:texmf-project'
resolvers       | files | file locator 'selfautoparent:texmf-project' not found
resolvers       | resolving | locating list of 'selfautoparent:texmf-fonts' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument 'selfautoparent:texmf-fonts'
resolvers       | files | file locator 'selfautoparent:texmf-fonts' not found
resolvers       | resolving | locating list of '/nix/store' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store'
resolvers       | files | file locator '/nix/store' found as '/nix/store'
resolvers       | resolving | locating list of '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf' (cached)
resolvers       | methods | resolving, method 'locators', how 'uri', handler 'file', argument '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf'
resolvers       | files | file locator '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf' found as '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf'
resolvers       | resolving |
resolvers       | methods | resolving, method 'generators', how 'uri', handler 'file', argument '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-var'
resolvers       | expansions | scanning path '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-var', branch '/nix/store/snzma9hi59yas7srynwiw8vsw9s8y7ac-texlive-combined-2015/share/texmf-var'
resolvers       | expansions | 35 files found on 6 directories with 1 uppercase remappings
resolvers       | methods | resolving, method 'generators', how 'uri', handler 'file', argument '/nix/store'
resolvers       | expansions | scanning path '/nix/store', branch '/nix/store'
~~~


Also noticed to following error during the environement build which does not
seem to have stopped it:

~~~
fmtutil: fmtutil is using the following fmtutil.cnf files (in precedence order):
fmtutil:   /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf/web2c/fmtutil.cnf
fmtutil: fmtutil is using the following fmtutil.cnf file for writing changes:
fmtutil:   /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-config/web2c/fmtutil.cnf
fmtutil: running `xetex -ini  -jobname=xetex -progname=xetex -etex xetex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/xetex/xetex.fmt installed.
fmtutil: running `pdftex -ini  -jobname=pdftex -progname=pdftex -translate-file=cp227.tcx *pdfetex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/pdftex/pdftex.fmt installed.
fmtutil: running `aleph -ini  -jobname=aleph -progname=aleph *aleph.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/aleph/aleph.fmt installed.
fmtutil: running `tex -ini  -jobname=tex -progname=tex tex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/tex/tex.fmt installed.
fmtutil: running `luatex -ini  -jobname=luatex -progname=luatex luatex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/luatex/luatex.fmt installed.
fmtutil: running `pdftex -ini  -jobname=pdfetex -progname=pdfetex -translate-file=cp227.tcx *pdfetex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/pdftex/pdfetex.fmt installed.
fmtutil: running `pdftex -ini  -jobname=mptopdf -progname=context -translate-file=cp227.tcx mptopdf.tex' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/pdftex/mptopdf.fmt installed.
fmtutil: running `xetex -ini  -jobname=cont-en -progname=context -8bit *cont-en.mkii' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/xetex/cont-en.fmt installed.
fmtutil: running `pdftex -ini  -jobname=cont-en -progname=context -8bit *cont-en.mkii' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/pdftex/cont-en.fmt installed.
fmtutil: running `pdftex -ini  -jobname=etex -progname=etex -translate-file=cp227.tcx *etex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/pdftex/etex.fmt installed.
fmtutil: running `pdftex -ini  -jobname=mllatex -progname=mllatex -translate-file=cp227.tcx -mltex *mllatex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/pdftex/mllatex.fmt installed.
fmtutil: running `xetex -ini  -jobname=xelatex -progname=xelatex -etex xelatex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/xetex/xelatex.fmt installed.
fmtutil: running `luatex -ini  -jobname=lualatex -progname=lualatex lualatex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/luatex/lualatex.fmt installed.
fmtutil: running `pdftex -ini  -jobname=latex -progname=latex -translate-file=cp227.tcx *latex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/pdftex/latex.fmt installed.
fmtutil: running `pdftex -ini  -jobname=pdflatex -progname=pdflatex -translate-file=cp227.tcx *pdflatex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/pdftex/pdflatex.fmt installed.
fmtutil: running `luatex -ini  -jobname=dviluatex -progname=dviluatex dviluatex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/luatex/dviluatex.fmt installed.
fmtutil: running `luatex -ini  -jobname=dvilualatex -progname=dvilualatex dvifmtutil [WARNING]: inifile euptex.ini for euptex/euptex not found.
fmtutil [WARNING]: inifile eptex.ini for eptex/eptex not found.
fmtutil [WARNING]: inifile ptex.ini for ptex/ptex not found.
fmtutil [WARNING]: inifile uptex.ini for uptex/uptex not found.
fmtutil [ERROR]: not building luajittex due to missing engine luajittex.
fmtutil [WARNING]: inifile cslatex.ini for cslatex/pdftex not found.
fmtutil [WARNING]: inifile utf8mex.ini for utf8mex/pdftex not found.
fmtutil [WARNING]: inifile xmltex.ini for xmltex/pdftex not found.
fmtutil [WARNING]: inifile csplain-utf8.ini for csplain/pdftex not found.
fmtutil [WARNING]: inifile texsis.ini for texsis/pdftex not found.
fmtutil [WARNING]: inifile amstex.ini for amstex/pdftex not found.
fmtutil [WARNING]: inifile mltex.ini for mltex/pdftex not found.
fmtutil [WARNING]: inifile platex.ini for platex/eptex not found.
fmtutil [WARNING]: inifile pdfxmltex.ini for pdfxmltex/pdftex not found.
fmtutil [WARNING]: inifile lambda.ini for lamed/aleph not found.
fmtutil [WARNING]: inifile mf.ini for mf/mf-nowin not found.
fmtutil [WARNING]: inifile cslatex.ini for pdfcslatex/pdftex not found.
fmtutil [WARNING]: inifile csplain.ini for pdfcsplain/luatex not found.
fmtutil [WARNING]: inifile csplain.ini for pdfcsplain/xetex not found.
fmtutil [WARNING]: inifile csplain-utf8.ini for pdfcsplain/pdftex not found.
fmtutil [WARNING]: inifile jadetex.ini for jadetex/pdftex not found.
fmtutil [WARNING]: inifile lollipop.ini for lollipop/pdftex not found.
fmtutil [WARNING]: inifile xelollipop.ini for xelollipop/xetex not found.
fmtutil [WARNING]: inifile pdfjadetex.ini for pdfjadetex/pdftex not found.
fmtutil [WARNING]: inifile mex.ini for mex/pdftex not found.
fmtutil [WARNING]: inifile pdfmex.ini for pdfmex/pdftex not found.
fmtutil [WARNING]: inifile uplatex.ini for uplatex/euptex not found.
fmtutil [WARNING]: inifile eplain.ini for eplain/pdftex not found.
fmtutil [WARNING]: inifile lualollipop.ini for lualollipop/luatex not found.
lualatex.ini' ...
fmtutil [INFO]: /nix/store/wj0wjsm8qywdkb53n1dlwnnng1ayhz01-texlive-combined-2015/share/texmf-var/web2c/luatex/dvilualatex.fmt installed.
fmtutil [INFO]: Disabled formats: 10
fmtutil [INFO]: Successfully rebuild formats: 17
~~~


## Technical details

* System: NixOS: `15.09.git.1f761bc (Dingo)`
* Nix version: `nix-env (Nix) 1.10`
* Nixpkgs version: See above

Restoring the following removed build environment variables bring back the missing
`texmf-var` directory:

~~~
    export TEXMFCNF="$out/share/texmf/web2c"
    export TEXMFDIST="$out/share/texmf"
    export TEXMFSYSCONFIG="$out/share/texmf-config"
    export TEXMFSYSVAR="$out/share/texmf-var"
~~~

But now, I'm back with this hanging at 

`resolvers       | expansions | scanning path '/nix/store', branch '/nix/store'^CMakefile:5: recipe for target 'resume.pdf' failed
make: *** [resume.pdf] Interrupt`

most likely resulting from:

`resolvers       | files | file locator '/nix/store/i2k1jyb5iq8515i737hdq9wfb922asgg-texlive-combined-2015/share/texmf-config' not found`

Here are the steps of my failed attempt at fixing the problem:

https://github.com/jraygauthier/nixpkgs/compare/jrg/master_user_20160122...jraygauthier:jrg/attempt_at_fixing_context_by_patching_luascript_as_well

There should be a way to stop this parent dir search at some point so that
this process does not attempt to search the content of `/nix/store`.

--- a/pkgs/tools/typesetting/tex/texlive-new/combine.nix
+++ b/pkgs/tools/typesetting/tex/texlive-new/combine.nix
@@ -60,21 +60,24 @@ in buildEnv {
     +
   ''
     export PATH="$out/bin:$out/share/texmf/scripts/texlive:${perl}/bin:$PATH"
-    export TEXMFCNF="$out/share/texmf/web2c"
-    export TEXMFDIST="$out/share/texmf"
-    export TEXMFSYSCONFIG="$out/share/texmf-config"
-    export TEXMFSYSVAR="$out/share/texmf-var"
     export PERL5LIB="$out/share/texmf/scripts/texlive"
   '' +
-    # patch texmf-dist -> texmf to be sure
-    # TODO: cleanup the search paths incl. SELFAUTOLOC, and perhaps do lua actions?
+    # patch texmf-{dist,local} -> texmf to be sure
+    # TODO: perhaps do lua actions?
     # tried inspiration from install-tl, sub do_texmf_cnf
   ''
     (
       cd ./share/texmf/web2c/
       local cnfOrig="$(realpath ./texmf.cnf)"
       rm ./texmf.cnf
-      cat "$cnfOrig" | sed 's/texmf-dist/texmf/g' > ./texmf.cnf
+      sed \
+        -e 's,texmf-dist,texmf,g' \
+        -e 's,texmf-local,texmf,g' \
+        -e "s,\$SELFAUTOLOC,$out,g" \
+        -e "s,\$SELFAUTODIR,$out/share,g" \
+        -e "s,\$SELFAUTOPARENT,$out/share,g" \
+        -e "s,\$SELFAUTOGRANDPARENT,$out/share,g" \
+        "$cnfOrig" > ./texmf.cnf
 
       rm updmap.cfg
     )
@@ -112,10 +115,6 @@ in buildEnv {
       rm "$link"
       makeWrapper "$target" "$link" \
         --prefix PATH : "$out/bin:${perl}/bin" \
-        --set TEXMFCNF "$out/share/texmf/web2c" \
-        --set TEXMFDIST "$out/share/texmf" \
-        --set TEXMFSYSCONFIG "$out/share/texmf-config" \
-        --set TEXMFSYSVAR "$out/share/texmf-var" \
         --prefix PERL5LIB : "$out/share/texmf/scripts/texlive"
 
       # avoid using non-nix shebang in $target by calling interpreter

--- a/pkgs/tools/typesetting/tex/texlive-new/combine.nix
+++ b/pkgs/tools/typesetting/tex/texlive-new/combine.nix
@@ -79,6 +79,19 @@ in buildEnv {
         -e "s,\$SELFAUTOGRANDPARENT,$out/share,g" \
         "$cnfOrig" > ./texmf.cnf
 
+      local cnfLuaOrig="$(realpath ./texmfcnf.lua)"
+      rm ./texmfcnf.lua
+      sed \
+        -e 's,texmf-dist,texmf,g' \
+        -e 's,texmf-local,texmf,g' \
+        -e "s,\(TEXMFLOCAL[ ]*=[ ]*\)[^\,]*,\1\"$out/share/texmf\",g" \
+        -e "s,\$SELFAUTOLOC,$out,g" \
+        -e "s,selfautodir:/,$out/share/,g" \
+        -e "s,selfautodir:,$out/share/,g" \
+        -e "s,selfautoparent:/,$out/share/,g" \
+        -e "s,selfautoparent:,$out/share/,g" \
+        "$cnfLuaOrig" > ./texmfcnf.lua
+
       rm updmap.cfg
     )
   '' +

